src/instana/options.py

```diff
@@ -13,57 +13,79 @@
     - AWSFargateOptions - Options class for AWS Fargate.  Holds settings specific to AWS Fargate.
     - GCROptions - Options class for Google cloud Run.  Holds settings specific to GCR.
 """
+
 import os
 import logging
+from typing import Any, Dict
+
+from instana.collector.helpers.runtime import parse_ignored_endpoints
 
-from .log import logger
-from .util.runtime import determine_service_name
+from instana.log import logger
+from instana.util.runtime import determine_service_name
+from instana.configurator import config
 
 
 class BaseOptions(object):
-    """ Base class for all option classes.  Holds items common to all """
+    """Base class for all option classes.  Holds items common to all"""
 
-    def __init__(self, **kwds):
+    def __init__(self, **kwds: Dict[str, Any]) -> None:
         self.debug = False
         self.log_level = logging.WARN
         self.service_name = determine_service_name()
         self.extra_http_headers = None
         self.allow_exit_as_root = False
+        self.ignore_endpoints = {}
 
         if "INSTANA_DEBUG" in os.environ:
             self.log_level = logging.DEBUG
             self.debug = True
 
         if "INSTANA_EXTRA_HTTP_HEADERS" in os.environ:
-            self.extra_http_headers = str(os.environ["INSTANA_EXTRA_HTTP_HEADERS"]).lower().split(';')
+            self.extra_http_headers = (
+                str(os.environ["INSTANA_EXTRA_HTTP_HEADERS"]).lower().split(";")
+            )
+
+        if "INSTANA_IGNORE_ENDPOINTS" in os.environ:
+            self.ignore_endpoints = parse_ignored_endpoints(
+                os.environ["INSTANA_IGNORE_ENDPOINTS"]
+            )
+        else:
+            if "tracing" in config:
+                if "ignore_endpoints" in config["tracing"]:
```

Comment:
```suggestion
            if "tracing" in config and "ignore_endpoints" in config["tracing"]:
```

Reply:
The inner if block is used to prevent errors on misuse of config["tracing"] as you can see below:
```python
>>> config = {"tracing": 15}
>>> "endpoint" in config["tracing"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: argument of type 'int' is not iterable
```

Reply:
I don't see differences from what was suggested:

```python
>>> config = {"tracing": 15}
>>>
>>> "tracing" in config and "ignore_endpoints" in config["tracing"]
Traceback (most recent call last):
  File "<python-input-3>", line 1, in <module>
    "tracing" in config and "ignore_endpoints" in config["tracing"]
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'int' is not iterable
>>>
>>>
>>> config = {"bla": "ble"}
>>> "tracing" in config and "ignore_endpoints" in config["tracing"]
False
>>>
```
If you expect a specific type of value and want to avoid user misconfiguration, you should use `isinstance(config["tracing"], expected_type)` or handle the `TypeError` exception in a try-except block.